+bids/query.m

```diff
@@ -216,11 +219,16 @@
                 result{end + 1} = fullfile(BIDS.subjects(i).path, modalities{j}, d(k).filename);
               end
 
-            case 'metadata'
+            case 'metafiles'
               if isfield(d(k), 'filename')
+                fmeta = BIDS.subjects(i).(modalities{j})(k).metafile;
```

Comment:
Don't see point of line 223

Reply:
Yup you are correct some of those IF blocks are most likely to always turn out to be true.
Just removed them.